### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by regislebrun**
This PR fixes issues #1843 (wrong default value for theta in AliMikhailHaqCopula documentation) and issue #1844 (bug in MaximumDistribution::computePDF()).
It also provides a fix for other discrepancy between the documentation of default copula parameters and the actual code.
Two evolutions are proposed for discussion:
  * the MemoizeFunction now propagates the MemoizeEvaluation through gradient and hessian if they are computed using finite differences
  * the ParametricFunction now computes the gradient() and parameterGradient() methods using only the needed part of the full gradient

The implementation is open to discussion because it introduces the special case of finite differences in several classes, which is the sign of a bad design. Nevertheless, the gain in terms of number of evaluations is worth considering.

### CODE REVIEW

**Comment by mbaudin47** [comment]
This should fix #1847.

File: ChangeLog [hunk 1]
```diff
@@ -20,6 +20,9 @@
 === Miscellaneous ===
 
 === Bug fixes ===
+ * #1843 (Ali-Mikhail-Haq copula parameter value)
+ * #1844 (MaximumDistribution::computePDF is wrong when all the marginals are equal and independent)
```
Comment (jschueller) [inline ChangeLog:24]: there's also #1845

File: python/test/t_ParametricFunction_std.py [hunk 2]
```diff
@@ -37,3 +38,50 @@
...
+ot_f = ot.MemoizeFunction(ot.PythonFunction(3, 3, py_f))
+param_f = ot.ParametricFunction(ot_f, [0, 1], [1.0, 2.0])
+
+x = [3.0]
+y = [1.5]
+n_calls_0 = ot_f.getCallsNumber()
+par_grad = param_f.parameterGradient(x)
+n_calls_1 = ot_f.getCallsNumber()
+assert_almost_equal(n_calls_1 - n_calls_0, 4, 0.0, 0.0)
+assert_almost_equal(n_calls, 4, 0.0, 0.0)
```
Comment (jschueller) [inline python/test/t_ParametricFunction_std.py:59]: it makes no sense to use assert_almost_equal for ints, just use `assert n_calls == 4, "message"`

**Review by jschueller — APPROVED** [review]
the parametric & memoize changes are ok

**Review by sofianehaddad — APPROVED** [review]

**Pull request merged**